x-pack/plugins/uptime/public/components/monitor/ping_list/columns/ping_timestamp/step_image_popover.tsx

```diff
@@ -82,13 +82,12 @@ export interface StepImagePopoverProps {
 
 const StepImageComponent: React.FC<
   Omit<StepImagePopoverProps, 'isImagePopoverOpen'> & {
-    setUrl: React.Dispatch<string | undefined>;
-    url: string | undefined;
+    setImageData: React.Dispatch<string | undefined>;
+    imageData: string | undefined;
   }
 > = (props) => {
   if (props.imgSrc) {
-    props.setUrl(props.imgSrc);
-    return <DefaultImage {...props} url={props.url} />;
+    return <DefaultImage {...props} />;
   } else if (props.imgRef) {
     return <RecomposedScreenshotImage {...props} imgRef={props.imgRef} />;
```

Comment:
Nit, but I think it would be easier to follow this logic if the props weren't spread wholesale, but if the prop were explicitly listed out.

Reply:
I think you're right. See https://github.com/elastic/kibana/pull/101678/commits/5952c1bc5efdd8fa3ce25f6637a6e1a90c6c6dfb.